### ISSUE TITLE
Reconstruction of non-square holograms raises error

### ISSUE THREAD

**Issue by LaurentRDC** [bmorris3/shampoo#15]
Reconstructing holograms with shape that is non-square yields a ValueError from the Hologram.apodize method.

Below is a script that reproduces the error, using data from the repository

```
import numpy as np
import os.path
from shampoo import Hologram

FILEPATH = 'USAF_test.tif'

if __name__ == '__main__':

    hologram = Hologram.from_tif(FILEPATH)
    original_shape = hologram.hologram.shape
    print('Image shape is square: {}'.format(original_shape))

    reshaped_data = np.resize(hologram.hologram, new_shape = (original_shape[0], original_shape[1] - 10))
    reshaped_hologram = Hologram(reshaped_data)
    print('New shape is not square: {}'.format(reshaped_data.shape))

    reconstructed = reshaped_hologram.reconstruct(propagation_distance = 0.03865)
```

This is a problem because some Allied Vision cameras have a resolution of 2550 x 2548

**Comment by bmorris3** [comment]
That's an interesting problem. Are the images actually asymmetric, or is there an additional overscan region that should be cropped off each image?

**Comment by bmorris3** [comment]
Can you envision any future circumstances when the field of view would be significantly non-square, and such a cropping would be detrimental? In the mean time, I'll implement a fix that crops the hologram to a square the size of the smallest dimension.

**Comment by LaurentRDC** [comment]
I think the reconstruction algorithm should be kept as general as possible. Maybe include in the documentation that the holograms must be square?

On my end I will make sure that cameras only output square images.

**Comment by bmorris3** [comment]
I agree that it should be kept general, but I'd like to note here that some FFT implementations are fastest when the dimensions of the matrix are 2^n (where n is an integer), so cropping the camera output to 2048 would be ideal for speed also.

**Comment by jkentwallace** [comment]
Laurent, squaring up the arrays (ideally to a 2^n size -> 2048x2048) is a good short term fix. 

If we are interested in keeping all of the data (instead of cropping) we can consider padding the raw array size up to the next convenient 2^n (for instance 4096). There are computational costs associated with this.

**Comment by bmorris3** [comment]
I've implemented a fix in #16. Once the continuous integration tests pass, I'll merge the PR.

@LaurentRDC – I have no doubts that you can figure out how to pull this commit onto your fork, but if you need any help, let me know.

**Comment by LaurentRDC** [comment]
Got it, thanks!
